js/angular/services/foundation.core.animation.js

```diff
@@ -56,16 +56,21 @@
       element[0].style.transitionDuration = '';
       element.addClass(activeClass);
 
-      element.one(events.join(' '), function() {
-        finishAnimation();
-      });
+      element.on(events.join(' '), eventHandler);
 
-      setTimeout(function() {
+      var animationTimeout = setTimeout(function() {
         if(timedOut) {
           finishAnimation();
         }
       }, 3000);
 
+      function eventHandler(e) {
+        if (element[0].id === e.target.id) {
```

Comment:
Wouldn't it be more verbose not to rely on the ```id``` but to use the comparison ```this === e.target``` as proposed in the jQuery docs (http://api.jquery.com/event.target/)?
> It is often useful to compare event.target to this in order to determine if the event is being handled due to event bubbling.